docs(rocprofiler-sdk): add 1.3.5 CHANGELOG entries (#10159)
ROCm/rocm-systems: projects/rocprofiler-sdk/CHANGELOG.md

    - `rocshmem_putmem_on_stream`, `rocshmem_getmem_on_stream`, and `rocshmem_alltoallmem_on_stream` are intercepted and emitted as per-call trace records.
    - Records are interleaved with HIP, HSA, RCCL, and other runtime traces so rocSHMEM communication activity can be viewed on the same timeline as GPU compute.
  - hipFile API tracing as a first-class tracing domain:
    - hipFile API calls are intercepted through dispatch-table wrapping and emitted as per-call trace records alongside HIP, HSA, and other runtime activity.
    - Enables file I/O operations to be correlated with GPU kernels and memory copies in a single profiling timeline.
  - Streaming Performance Monitor (SPM) counter data in the rocpd output format:
    - SPM records are stored as `rocpd_track` rows labelled `SPM`, with counter values grouped by timestamp into `rocpd_sample` rows and per-dimension data in `rocpd_pmc_event` rows.
    - The rocpd schema gains the `sample_id`, `xcc`, `shader_engine`, and `instance` columns.
    - SPM data is consumable by any tool that reads the rocpd database and convertible to the other output formats, such as Perfetto.

>   - Streaming Performance Monitor (SPM) counter collection support (beta):

Reviewer: This section also needs to be moved to  ROCprofiler-SDK 1.3.2

## ROCprofiler-SDK 1.3.2 for ROCm release 7.14

### Added

**API:**

  - SPM counter collection support in `rocprofv3` (beta):
    - `--spm <counter>` flag to specify counters for SPM collection.
    - `--spm-sample-interval` and `--spm-sample-interval-unit` parameters to configure sampling rate.
    - `--spm-beta-enabled` flag to opt in to the beta SPM feature.
    - `--spm-config` option in `rocprofv3-avail` to list available SPM configurations.
  - JSON and rocpd output format support for SPM.
  - OpenMP (OMPT) tracing via the new `--ompt-trace` flag:
    - Accepts a bare boolean or category list (`all, thread, parallel, task, sync, mutex, target, device, error`); also folded into `--sys-trace`/`--runtime-trace`.
    - rocpd-only trace: records go to the rocpd database (the default output format) and export via `rocpd convert`.

**Documentation:**

  - SPM API reference guide (`api-reference/spm.rst`).
  - SPM usage guide for `rocprofv3` (`how-to/using-spm.rst`).
  - `--spm-config` documentation to `rocprofv3-avail` usage guide.

### Changed

> - Bump rocpd schema to version 3.0.1 which supports NIC agent types.

Reviewer: Why is this removed altogether?

Author: its not removed. Its in Unreleased section

>   - JSON and rocpd output format support for SPM.

Reviewer: remove rocpd here

### Removed

## ROCprofiler-SDK 1.3.5 for ROCm release 10.0.0

### Added

**API:**

  - rocSHMEM host-stream API interception for the rocSHMEM tracing domain introduced in 1.3.0:
    - `rocshmem_putmem_on_stream`, `rocshmem_getmem_on_stream`, and `rocshmem_alltoallmem_on_stream` are intercepted and emitted as per-call trace records.
    - Records are interleaved with HIP, HSA, RCCL, and other runtime traces so rocSHMEM communication activity can be viewed on the same timeline as GPU compute.
  - hipFile API tracing as a first-class tracing domain:
    - hipFile API calls are intercepted through dispatch-table wrapping and emitted as per-call trace records alongside HIP, HSA, and other runtime activity.
    - Enables file I/O operations to be correlated with GPU kernels and memory copies in a single profiling timeline.
  - Streaming Performance Monitor (SPM) counter data in the rocpd output format:
    - SPM records are stored as `rocpd_track` rows labelled `SPM`, with counter values grouped by timestamp into `rocpd_sample` rows and per-dimension data in `rocpd_pmc_event` rows.
    - The rocpd schema gains the `sample_id`, `xcc`, `shader_engine`, and `instance` columns.

>     - SPM data is consumable by any tool that reads the rocpd database and convertible to the other output formats, such as Perfetto.

Reviewer: SPM  rocpd output is only convertible to CSV format.

Author: done

  - Advanced Thread Trace (ATT) support in the live attach workflow:
    - On attach, the SDK registers for code-object iteration and creation callbacks so thread trace operates correctly on code objects that were loaded before the attach occurred.
    - Makes ATT usable on already-running production workloads without an application restart.
  - Experimental SQTT quick scan mode for thread trace, enabled through a new CMake flag:
    - Collects thread trace data without packet insertion or HSA signal manipulation, removing the queue interception overhead required by the standard ATT path.
    - Individual kernels can be traced without serialization, and the path is independent of the ROCm runtime version.
    - Experimental: intended to validate the new collection path and to enable out-of-process thread trace and long-kernel tracing in future releases.

> **rocprofv3 (CLI):**

Reviewer 2: We also have --att-no-detail now.

  - OpenMP (OMPT) tracing via the new `--ompt-trace` flag:
    - Accepts a bare boolean or a space-separated category list (`all, thread, parallel, task, sync, mutex, target, device, error`), following the same style as `--pmc` and `--output-format`; for example, `--ompt-trace parallel task target sync`. Also folded into `--sys-trace`/`--runtime-trace`.

>     - rocpd-only trace: records go to the rocpd database (the default output format) and export via `rocpd convert`.

Reviewer 3: ```suggestion
    - rocpd-only trace: records go to the rocpd database (the default output format) and are exported via `rocpd convert`.
```

Author: done

>     - The OMPT callback layer has been supported by ROCprofiler-SDK since an earlier release; this flag makes it accessible without writing a custom tool.

Reviewer 3: ```suggestion
    - The OMPT callback layer is already supported by ROCprofiler-SDK; this flag makes it accessible without writing a custom tool.
```

Author: done

    - Wraps the decoder library and exposes thread trace decoding as a first-class Python interface, with samples demonstrating common workflows.
    - Useful for analysis scripts, Jupyter notebooks, and custom profiling tools that process ATT output programmatically.
    - Decoder integration tests have been migrated to Python, simplifying test authoring and making it easier for downstream tools to validate their trace-decoding pipelines.

### Changed

- Previously, `rocattach` calculated symbol offsets from the host's `librocprofiler-register.so` and applied them to the target's mapping, which failed when the host and container libraries differ in ELF layout or path. Offsets are now resolved from the target process itself.
- Idle inline queues with no active profiling consumers now bypass queue interposition entirely, reducing overhead for applications that create queues but do not immediately dispatch work.
- DWARF information is now parsed lazily, reducing startup overhead for attach and tracing sessions on large binaries.
- Disabled autoflush in thread trace to prevent premature buffer flushes that produced incomplete or corrupted traces.

> ### Resolved issues

Reviewer 2: Hi: Another resolved issue recently is external aqlprofile was being loaded and could cause failures besides not being used.
